packages/front-end/components/Settings/DataSourceForm.tsx

```diff
@@ -142,10 +142,21 @@ const DataSourceForm: FC<{
   data: Partial<DataSourceInterfaceWithParams>;
   existing: boolean;
   source: string;
-  onCancel: () => void;
-  onSuccess: (id: string) => Promise<void>;
+  onCancel?: () => void;
+  onSuccess: (id?: string) => Promise<void>;
```

Comment:
Revert this line and keep the `id` argument as required.  It can just be ignored if not needed.